File: solutions/rb/30-12926.rb
def solution(s, n)
  s.split(/ /,-1).map {|w| w.chars.map {|c| (c.ord+n > (/[[:upper:]]/.match(c) ? 90 : 122) ? c.ord+n-26 : c.ord+n).chr("UTF-8") }.join}.join(' ')
end

p solution('AB', 1)
p solution('z', 1)
p solution(' a  B z ', 4)

File: solutions/py/30-12926.py
def solution(s, n):
    result = ''
    for char in s:
        if char != ' ':
            max_range = ord('Z') if char.isupper() else ord('z')
            d = n - 26 if ord(char) + n > max_range else n
            char = chr(ord(char)+d)
        result += char
    return result


print(solution('AB', 1)) #=> 'BC'
print(solution('z', 1)) #=> 'a'
print(solution('a B z', 4)) #=> 'e F d'
